File: src/test/java/no/poc/domain/DomainBuilderTest.java
package no.poc.domain;

import org.junit.Test;

import static no.poc.domain.builder.BrukerBuilder.aBruker;
import static no.poc.domain.builder.KravBuilder.aKrav;
import static no.poc.domain.builder.SakBuilder.aSak;
import static no.poc.domain.builder.TypeBuilder.aBuilderOfType;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class DomainBuilderTest {
    @Test
    public void buildDomain() {
        Sak sak = aSak()
                .withSakId(1L)
                .with(aBruker().ofTypeAndebyBruker())
                .with(aKrav()
                        .withKravId(1L)
                        .with(aBuilderOfType(KravType.class).withTypeId("TYPE_1").withName("Type 1"))
                )
                .with(aKrav()
                        .withKravId(2L)
                        .with(aBuilderOfType(KravType.class).withTypeId("TYPE_2").withName("Type 2"))
                )
                .build();

        assertThat(sak.getSakId(), is(1L));
        assertThat(sak.getBruker().getBrukerId(), is(1L));
        assertThat(sak.getBruker().getFornavn(), is("Donald"));
        assertThat(sak.getBruker().getEtternavn(), is("Duck"));
        assertThat(sak.getKravList().size(), is(2));
        assertThat(sak.getKravList().get(0).getKravId(), is(1L));
        assertThat(sak.getKravList().get(0).getKravType().getKravTypeId(), is("TYPE_1"));
        assertThat(sak.getKravList().get(0).getKravType().getKravTypeNavn(), is("Type 1"));
        assertThat(sak.getKravList().get(1).getKravId(), is(2L));
        assertThat(sak.getKravList().get(1).getKravType().getKravTypeId(), is("TYPE_2"));
        assertThat(sak.getKravList().get(1).getKravType().getKravTypeNavn(), is("Type 2"));
    }
}

File: src/test/java/no/poc/domain/DomainBuilderTest.kt
package no.poc.domain

import no.poc.domain.builder.BrukerBuilder.aBruker
import no.poc.domain.builder.KravBuilder.aKrav
import no.poc.domain.builder.SakBuilder.aSak
import no.poc.domain.builder.TypeBuilder.aBuilderOfType
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class DomainBuilderTest {

    @Test
    fun buildDomain() {
        val sak = aSak()
                .withSakId(1L)
                .with(aBruker().ofTypeAndebyBruker())
                .with(aKrav()
                        .withId(1L)
                        .with(aBuilderOfType(KravType::class.java).withTypeId("TYPE_1").withName("Type 1"))
                )
                .with(aKrav()
                        .withId(2L)
                        .with(aBuilderOfType(KravType::class.java).withTypeId("TYPE_2").withName("Type 2"))
                        .with(aBruker().withId(2L).withOrganisasjonsnavn("Bedrift AS"))
                )
                .build()

        assertThat(sak.bruker.id).isEqualTo(1L)
        assertThat(sak.bruker.fornavn).isEqualTo("Donald")
        assertThat(sak.bruker.etternavn).isEqualTo("Duck")

        assertThat(sak.kravList).hasSize(2)

        assertThat(sak.kravList[0].id).isEqualTo(1L)
        assertThat(sak.kravList[0].kravType.kravTypeId).isEqualTo("TYPE_1")
        assertThat(sak.kravList[0].kravType.kravTypeNavn).isEqualTo("Type 1")
        assertThat(sak.kravList[0].bruker).isEmpty

        assertThat(sak.kravList[1].id).isEqualTo(2L)
        assertThat(sak.kravList[1].kravType.kravTypeId).isEqualTo("TYPE_2")
        assertThat(sak.kravList[1].kravType.kravTypeNavn).isEqualTo("Type 2")
        assertThat(sak.kravList[1].bruker).isPresent
        assertThat(sak.kravList[1].bruker.get().organisasjonsnavn).isEqualTo("Bedrift AS")
        assertThat(sak.kravList[1].bruker.get().isBedrift()).isTrue()
    }
}